sphinx/writers/latex.py

```diff
@@ -1444,7 +1445,9 @@ def visit_entry(self, node):
             self.needs_linetrimming = 1
         if len(node.traverse(nodes.paragraph)) >= 2:
             self.table.has_oldproblematic = True
-        if isinstance(node.parent.parent, nodes.thead):
+        if isinstance(node.parent.parent, nodes.thead) or \
+           node.parent.parent.parent.stubs[cell.col]:
+            # "stubs" list is an attribute of the tgroup element
             if len(node) == 1 and isinstance(node[0], nodes.paragraph) and node.astext() == '':
                 pass
             else:
```

Comment:
Done at e02f3f1 for 1.6 and ``\sphinxstylethead`` removed at 4fc7702 for 1.7 (the `\unskip` is now unneeded because the braces are gone and end of table cell does ``\unskip`` automatically to remove trailing space). By the way, I meant `\sffamily`, not `\ttfamily`...